### ISSUE TITLE
Error when printing all data, after creating bitcoin rawtx

### ISSUE THREAD

**Issue by andrea-dintino** [dyne/Zenroom#272]
The script: 

```
Given I have a 'testnet address' named 'sender'
Given I have a 'testnet address' named 'recipient'
Given I have a 'satoshi fee'
Given I have a 'satoshi amount'
Given I have a 'satoshi unspent'
Given I have the 'keys'
When I rename 'satoshi unspent' to 'testnet unspent'
When I create the testnet transaction
When I sign the testnet transaction
When I create the testnet raw transaction
When I create the size of 'testnet raw transaction'
Then print all data
# Then print the 'testnet raw transaction' as 'hex'
```

With the data

```
{
	"keys": {
		"testnet": "cMqN1QyWYZqAAw5qVH8AzFF3GVM9VQKyNEY7ZUQsPivVpVYWjzfx"
	},
	"satoshi unspent": [
		{
			"txid": "dd6d3c58fe0cd8729e731545830307fcdd36620c14bc5988308e0485d23ea53c",
			"vout": 1,
			"status": {
				"confirmed": true,
				"block_height": 2103197,
				"block_hash": "0000000000000004e0ff305a04c1b23e26cf4c6a903b2304eb58a8b18c81414d",
				"block_time": 1636388875
			},
			"value": 79716
		},
		{
			"txid": "f435e5f2139b7a919b51ee6950b82f8b60031158960d66ec67135320d68f54a2",
			"vout": 1,
			"status": {
				"confirmed": true,
				"block_height": 2102869,
				"block_hash": "000000000000f4cb5540d429d37e3716317acc34f071a90f14c8a4536716382e",
				"block_time": 1636187124
			},
			"value": 1563967
		},
		{
			"txid": "63145fe24a787a25e721e6be4fc3db08dbec21f7924d54d68d139c6506f509f7",
			"vout": 0,
			"status": {
				"confirmed": true,
				"block_height": 2102869,
				"block_hash": "000000000000f4cb5540d429d37e3716317acc34f071a90f14c8a4536716382e",
				"block_time": 1636187124
			},
			"value": 80000
		},
		{
			"txid": "98aea64e8c923240f8abc9af577bd68ecc5a20d9f6b1b6d8e174a57b68d825bf",
			"vout": 1,
			"status": {
				"confirmed": true,
				"block_height": 2103471,
				"block_hash": "00000000000000287d5251d6be9a8d8f18fdf91bb4e20a14471a025384d593a8",
				"block_time": 1636567249
			},
			"value": 96560
		},
		{
			"txid": "7166ff2dd8cd0893253c984c031896335d68f31d18aafffb1020f024735fc5c9",
			"vout": 1,
			"status": {
				"confirmed": true,
				"block_height": 2102869,
				"block_hash": "000000000000f4cb5540d429d37e3716317acc34f071a90f14c8a4536716382e",
				"block_time": 1636187124
			},
			"value": 100000
		},
		{
			"txid": "592c35a7e710828746988926317f6929073ece459b55f935e0a8091b97564f24",
			"vout": 1,
			"status": {
				"confirmed": true,
				"block_height": 2101106,
				"block_hash": "0000000000000020e76792ea88ad80ed6cf634b9bc08f0adc2cc7c8db5f22a5a",
				"block_time": 1635376667
			},
			"value": 995036
		},
		{
			"txid": "60431824fa7a2b085b7cabf72bbaa3cfe97ffe69708e0d8bccbe5c301eed30e5",
			"vout": 1,
			"status": {
				"confirmed": true,
				"block_height": 2103470,
				"block_hash": "0000000000003317181c3239426b7e9a1aed9164cb4726b2777ca50b2c54309d",
				"block_time": 1636566498
			},
			"value": 79716
		},
		{
			"txid": "17a37cdbab83718435a0bebd747337a943add4af34b9c124c14f679b5a0b1cd0",
			"vout": 0,
			"status": {
				"confirmed": true,
				"block_height": 2101201,
				"block_hash": "000000000001aad40c5be20f667630f2a85dc0da3003c4605bdb628ee180c325",
				"block_time": 1635436736
			},
			"value": 100000
		},
		{
			"txid": "3fcdad9b6289fbba25d306fadac78062949c63164eabd35e4541c3f84992ffb1",
			"vout": 12,
			"status": {
				"confirmed": true,
				"block_height": 2102981,
				"block_hash": "000000000000000f9e0f2e29a4756a7564a38eade977d5253271c41fa8de2760",
				"block_time": 1636256674
			},
			"value": 3527419
		}
	],
	"satoshi amount": "1",
	"satoshi fee": "141",
	"recipient": "tb1q73czlxl7us4s6num5sjlnq6r0yuf8uh5clr2tm",
	"sender": "tb1qc5wzp53l39v499nvycmcvu2aaqlu84xnkhq3dv"
}
```

Returns the error: 

[W] Zencode is missing version check, please add: rule check version N.N.N[W] **Then print all data 2[W] nil[W]** D ZEN:pick found sender[W] D ZEN:pick found recipient[W] D ZEN:pick found satoshi_fee[W] D ZEN:pick found satoshi_amount[W] D ZEN:pick found satoshi_unspent[W] D ZEN:pick found keys[W] [31;1m[!][0m **/zencode_bitcoin.lua:58: attempt to index a nil value (field 'address')**[W]

**Comment by andrea-dintino** [comment]
This also doesn't work:

```
Given nothing
When I create the bitcoin key
Then print my 'bitcoin' from 'keys' as 'hex'
```

**Comment by matteo-cristino** [comment]
The second example now works by adding your identity
```gherkin
Given I am 'Pippo'
When I create the bitcoin key
Then print my 'bitcoin' from 'keyring' as 'hex'
````

For the first one, after replacing `keys` with `keyring`, I get the following error
```
[!]  ERROR:
[!]  EXEC: [string "zenroom_common"]:175: Zenguard detected an invalid value in HEAP: amount (number)
````
Where `amount` corresponds to `value` found in `satoshi_unspent` in the data

**Comment by matteo-cristino** [comment]
Moreover the [zencode bitcoin tests](https://github.com/dyne/Zenroom/blob/master/test/zencode/bitcoin.bats) tests few things and uses only the testnet version of it.

**Comment by matteo-cristino** [comment]
Fixed in #679

Since `keys` is been replaced by `keyring` the example now is as follow:
* Zencode:
```gherkin
Given I have a 'testnet address' named 'sender'
Given I have a 'testnet address' named 'recipient'
Given I have a 'satoshi fee'
Given I have a 'satoshi amount'
Given I have a 'satoshi unspent'
Given I have the 'keyring'
When I rename 'satoshi unspent' to 'testnet unspent'
When I create the testnet transaction
When I sign the testnet transaction
When I create the testnet raw transaction
When I create the size of 'testnet raw transaction'
Then print all data
```
* data:
```json
{
	"keyring": {
		"testnet": "cMqN1QyWYZqAAw5qVH8AzFF3GVM9VQKyNEY7ZUQsPivVpVYWjzfx"
	},
	"satoshi unspent": [
		{
			"txid": "dd6d3c58fe0cd8729e731545830307fcdd36620c14bc5988308e0485d23ea53c",
			"vout": 1,
			"status": {
				"confirmed": true,
				"block_height": 2103197,
				"block_hash": "0000000000000004e0ff305a04c1b23e26cf4c6a903b2304eb58a8b18c81414d",
				"block_time": 1636388875
			},
			"value": 79716
		},
		{
			"txid": "f435e5f2139b7a919b51ee6950b82f8b60031158960d66ec67135320d68f54a2",
			"vout": 1,
			"status": {
				"confirmed": true,
				"block_height": 2102869,
				"block_hash": "000000000000f4cb5540d429d37e3716317acc34f071a90f14c8a4536716382e",
				"block_time": 1636187124
			},
			"value": 1563967
		},
		{
			"txid": "63145fe24a787a25e721e6be4fc3db08dbec21f7924d54d68d139c6506f509f7",
			"vout": 0,
			"status": {
				"confirmed": true,
				"block_height": 2102869,
				"block_hash": "000000000000f4cb5540d429d37e3716317acc34f071a90f14c8a4536716382e",
				"block_time": 1636187124
			},
			"value": 80000
		},
		{
			"txid": "98aea64e8c923240f8abc9af577bd68ecc5a20d9f6b1b6d8e174a57b68d825bf",
			"vout": 1,
			"status": {
				"confirmed": true,
				"block_height": 2103471,
				"block_hash": "00000000000000287d5251d6be9a8d8f18fdf91bb4e20a14471a025384d593a8",
				"block_time": 1636567249
			},
			"value": 96560
		},
		{
			"txid": "7166ff2dd8cd0893253c984c031896335d68f31d18aafffb1020f024735fc5c9",
			"vout": 1,
			"status": {
				"confirmed": true,
				"block_height": 2102869,
				"block_hash": "000000000000f4cb5540d429d37e3716317acc34f071a90f14c8a4536716382e",
				"block_time": 1636187124
			},
			"value": 100000
		},
		{
			"txid": "592c35a7e710828746988926317f6929073ece459b55f935e0a8091b97564f24",
			"vout": 1,
			"status": {
				"confirmed": true,
				"block_height": 2101106,
				"block_hash": "0000000000000020e76792ea88ad80ed6cf634b9bc08f0adc2cc7c8db5f22a5a",
				"block_time": 1635376667
			},
			"value": 995036
		},
		{
			"txid": "60431824fa7a2b085b7cabf72bbaa3cfe97ffe69708e0d8bccbe5c301eed30e5",
			"vout": 1,
			"status": {
				"confirmed": true,
				"block_height": 2103470,
				"block_hash": "0000000000003317181c3239426b7e9a1aed9164cb4726b2777ca50b2c54309d",
				"block_time": 1636566498
			},
			"value": 79716
		},
		{
			"txid": "17a37cdbab83718435a0bebd747337a943add4af34b9c124c14f679b5a0b1cd0",
			"vout": 0,
			"status": {
				"confirmed": true,
				"block_height": 2101201,
				"block_hash": "000000000001aad40c5be20f667630f2a85dc0da3003c4605bdb628ee180c325",
				"block_time": 1635436736
			},
			"value": 100000
		},
		{
			"txid": "3fcdad9b6289fbba25d306fadac78062949c63164eabd35e4541c3f84992ffb1",
			"vout": 12,
			"status": {
				"confirmed": true,
				"block_height": 2102981,
				"block_hash": "000000000000000f9e0f2e29a4756a7564a38eade977d5253271c41fa8de2760",
				"block_time": 1636256674
			},
			"value": 3527419
		}
	],
	"satoshi amount": "1",
	"satoshi fee": "141",
	"recipient": "tb1q73czlxl7us4s6num5sjlnq6r0yuf8uh5clr2tm",
	"sender": "tb1qc5wzp53l39v499nvycmcvu2aaqlu84xnkhq3dv"
}
```
